base/partr.jl

```diff
@@ -61,45 +43,76 @@ function sift_up(heap::taskheap, idx::Int32)
 end
 
 
-function sift_down(heap::taskheap, idx::Int32)
+function multiq_sift_down(heap::taskheap, idx::Int32)
     if idx <= heap.ntasks
         for child = (heap_d * idx - heap_d + Int32(2)):(heap_d * idx + Int32(1))
-            child > tasks_per_heap && break
+            child > length(heap.tasks) && break
             if isassigned(heap.tasks, Int(child)) &&
                     heap.tasks[child].priority < heap.tasks[idx].priority
                 t = heap.tasks[idx]
                 heap.tasks[idx] = heap.tasks[child]
                 heap.tasks[child] = t
-                sift_down(heap, child)
+                multiq_sift_down(heap, child)
             end
         end
     end
 end
 
 
+function multiq_size()
+    heap_c = UInt32(2)
+    heap_p = UInt32(length(heaps))
+    nt = UInt32(nthreads())
+
+    if heap_c * nt <= heap_p
```

Comment:
If `heap_p` is too much larger than `heap_c * nthreads` then `deletemin` delay will be larger than necessary. Suggest leaving a TODO comment here to work out how to resize downwards.

Reply:
I think we might want to alter the scaling to be `min(heap_c * nthreads, current_load)` where `current_load` is the sum of the length of all of the tasks queues (or estimated statistically from a small sample). But I haven't attempted any tuning yet, just focusing on the dynamic scaling for now.